elyra/metadata/metadata_app.py

```diff
@@ -376,6 +378,100 @@ def start(self):
+
+    schema_name_option = CliOption("--schema_name", name='schema_name',
+                                   description='The schema name of the metadata instances to export',
+                                   required=False)
+
+    valid_only_flag = Flag("--valid-only", name='valid-only',
+                           description='Only export valid instances (default includes invalid instances)',
+                           default_value=False)
+
+    clean_flag = Flag("--clean", name='clean',
+                      description='Clear out contents of the export directory',
+                      default_value=False)
+
+    directory_option = CliOption("--directory", name='directory',
+                                 description='The local file system path where the exported metadata will be stored',
+                                 required=True)
+
+    # 'Export' flags
+    options: List[Option] = [schema_name_option, valid_only_flag, clean_flag, directory_option]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.metadata_manager = MetadataManager(schemaspace=self.schemaspace)
+
+    def start(self):
+        super().start()  # process options
+
+        schema_name = self.schema_name_option.value
+        if(schema_name):
+            schema_list = sorted(list(self.schemas.keys()))
+            if schema_name not in schema_list:
+                print(f"Schema name '{schema_name}' is invalid. For the '{self.schemaspace}' schemaspace, " +
+                      f"the schema name must be one of {schema_list}")
+                self.exit(1)
+
+        include_invalid = not self.valid_only_flag.value
+        directory = self.directory_option.value
+        clean = self.clean_flag.value
+
+        try:
+            if(self.schema_name_option is not None):
+                metadata_instances = self.metadata_manager.get_all(include_invalid=include_invalid,
+                                                                   of_schema=schema_name)
+            else:
+                metadata_instances = self.metadata_manager.get_all(include_invalid=include_invalid)
+        except MetadataNotFoundError:
+            metadata_instances = None
+
+        if not metadata_instances:
+            print(f"No metadata instances found for schemaspace '{self.schemaspace}'" +
+                  (f" and schema '{schema_name}'" if schema_name else ""))
+            print(f"Nothing exported to '{directory}'")
+            return
+
+        dest_directory = os.path.join(directory, self.schemaspace)
+
+        if not os.path.exists(dest_directory):
+            try:
+                print(f"Creating directory structure for '{dest_directory}'")
+                os.makedirs(dest_directory)
+            except OSError as e:
+                print(f"Error creating directory structure for {dest_directory}: {e.strerror}: '{e.filename}'")
+                self.exit(1)
+        else:
```

Comment:
Moved the `if clean` portion to the `else` as there is nothing there to clean in a newly created directory